nmpolicy/internal/resolver/resolver.go

```diff
@@ -105,7 +106,7 @@ func (r *resolver) resolveCaptureEntryName(captureEntryName string) (types.NMSta
 	return capturedStateEntry.State, nil
 }
 
-func (r resolver) resolveCaptureASTEntry() (types.NMState, error) {
+func (r *resolver) resolveCaptureASTEntry() (types.NMState, error) {
```

Comment:
Why do we need to change this to pointer ? is better to keep as less pointer receiver as possible so we know at what receivers the resolver can be changed.

Reply:
Ahh right one of the delegated function do change the `resolver` state so this one has to be a pointer.